hips/tiles/tile.py

```diff
@@ -0,0 +1,117 @@
+
+
+class HipsTile:
+    """HiPS tile container.
+
+    This class provides methods for fetching, reading, and writing a HiPS tile.
+
+    Parameters
+    ----------
+    hips_tile_meta : `HipsTileMeta`
+        Metadata of HiPS tile
+    format : `str`
+        Format of HiPS tile
+    data : `np.ndarray`
+        Data containing HiPS tile
+    header : `astropy.io.fits.Header`
+        Header of HiPS tile
+
+    Examples
+    --------
+        >>> import urllib.request
+        >>> from hips.tiles import HipsTile
+        >>> from hips.tiles import HipsTileMeta
+        >>> from astropy.tests.helper import remote_data
+        >>> hips_tile_metadata = HipsTileMeta(order=6, ipix=30889, format='fits')
+        >>> hips_tile = HipsTile.read(hips_tile_metadata, 'Npix30889.fits')
+        >>> hips_tile.data
+        array([[0, 0, 0, ..., 0, 0, 0],
+               [0, 0, 0, ..., 0, 0, 0],
+               [0, 0, 0, ..., 0, 0, 0],
+               ...,
+               [0, 0, 0, ..., 1, 0, 0],
+               [0, 0, 0, ..., 1, 0, 1],
+               [0, 0, 0, ..., 1, 0, 1]], dtype=int16)
+        """
+
+    def __init__(self, hips_tile_meta: HipsTileMeta, data: np.ndarray = None, header: Header = None) -> None:
+        self.hips_tile_meta = hips_tile_meta
+        self.format = format
+        self.data = data
+        self.header = header
+
+    @classmethod
+    def fetch(cls, hips_tile_meta: HipsTileMeta, url: str) -> 'HipsTile':
+        """Fetch HiPS tile and load into memory (`HipsTile`).
+
+        Parameters
+        ----------
+        hips_tile_meta : `HipsTileMeta`
+            Metadata of HiPS tile
+        url : `str`
+            URL containing HiPS tile
+        """
+        raw_image = BytesIO(urllib.request.urlopen(url).read())
+        if hips_tile_meta.format == 'fits':
+            hdulist = fits.open(raw_image)
+            data = np.array(hdulist[0].data)
+            header = hdulist[0].header
+            return cls(hips_tile_meta, data, header)
+        else:
+            data = np.array(Image.open(raw_image))
+            return cls(hips_tile_meta, data)
+
+    @classmethod
+    def read(cls, hips_tile_meta: HipsTileMeta, filename: str) -> 'HipsTile':
+        """Read HiPS tile data from a directory and load into memory (`HipsTile`).
+
+        Parameters
+        ----------
+        hips_tile_meta : `HipsTileMeta`
+            Metadata of HiPS tile
+        filename : `str`
+            File name of HiPS tile
+        """
+        path = tile_path().absolute() / filename
+        if hips_tile_meta.format == 'fits':
+            hdulist = fits.open(path)
+            data = np.array(hdulist[0].data)
+            header = hdulist[0].header
+            return cls(hips_tile_meta, data, header)
+        else:
+            data = np.array(Image.open(str(path)))
+            return cls(hips_tile_meta, data)
+
+    def write(self, filename: str) -> None:
+        """Write HiPS tile by a given filename (`None`).
```

Comment:
Remove `(None)` here. (no need to mention this in docstrings that don't return anything.